staging/src/k8s.io/kube-aggregator/pkg/apiserver/resolvers.go

```diff
@@ -28,6 +28,11 @@ type ServiceResolver interface {
 	ResolveEndpoint(namespace, name string, port int32) (*url.URL, error)
 }
 
+// A ServiceHostnameResolver knows how to get the hostname of an external service.
+type ServiceHostnameResolver interface {
+	ResolveHostname(namespace, name string) (string, bool, error)
+}
```

Comment:
This is not the hostname, but the servername. "servername" is a fixed term in tls, aka "Server Name Indication" in SNI.

Reply:
renamed in https://github.com/kubernetes/kubernetes/pull/123808/commits/ad9c3025e757f3ba1cfc19547a8056031d390508